### ISSUE TITLE
 fatal error: fuse3/fuse.h No such file or directory

### ISSUE THREAD

**Issue by RobbeHogent** [sgan81/apfs-fuse#87]
I have downloaded the libfuse-dev package but I still get the error "fatal error: fuse3/fuse.h".
Does this mean I have to manually install the fuse3 library?
I thought it was included in the libfuse-dev package?

**Comment by sgan81** [comment]
Either use FUSE 2 (configure with ccmake .) or install FUSE 3, probably something like libfuse3-dev

**Comment by a-piece-of-garbage** [comment]
Install libfuse3-dev.

**Comment by RobbeHogent** [comment]
I had already tried to configure it with fuse 2 without success and now I tried to install libfuse3-dev but the package can't be found.

**Comment by Banaanhangwagen** [comment]
You might look over here:
https://github.com/libfuse/libfuse/releases

**Comment by sgan81** [comment]
You couldn't configure it for fuse 2? In the build directory, do the following:
1. Execute `cmake ..`
2. Execute `ccmake .`
3. Navigate to `USE_FUSE3`, and if it is not set to `OFF`, hit enter
4. Press `c` to configure
5. Press `g` to generate Makefile and exit ccmake
6. Execute `make`

**Comment by Iaxama** [comment]
> You couldn't configure it for fuse 2? In the build directory, do the following:
> 
> 1. Execute `cmake ..`
> 2. Execute `ccmake .`
> 3. Navigate to `USE_FUSE3`, and if it is not set to `OFF`, hit enter
> 4. Press `c` to configure
> 5. Press `g` to generate Makefile and exit ccmake
> 6. Execute `make`

This worked for me. Maybe that option should be set to OFF by default as it would not compile if you just follow the installation instructions

**Comment by AMDphreak** [comment]
problem still exists.

**Comment by AMDphreak** [comment]
> You couldn't configure it for fuse 2? In the build directory, do the following:
> 
>     1. Execute `cmake ..`
> 
>     2. Execute `ccmake .`
> 
>     3. Navigate to `USE_FUSE3`, and if it is not set to `OFF`, hit enter
> 
>     4. Press `c` to configure
> 
>     5. Press `g` to generate Makefile and exit ccmake
> 
>     6. Execute `make`

If by

    `c` to "configure" 

you they meant "write to file" then the ccmake app utterly failed to express their intentions in the command. Gotta love half-baked UI's.

**Comment by jivanpal** [comment]
> > [...]
> > 3. Navigate to `USE_FUSE3`, and if it is not set to `OFF`, hit enter
> > [...]
> 
> This worked for me. Maybe that option should be set to OFF by default as it would not compile if you just follow the installation instructions

Indeed, I also just had to do this on 64-bit Ubuntu 18.04, since `libfuse-dev` doesn't seem to install the FUSE3 headers. +1 for making `USE_FUSE3=OFF` the default.

**Comment by jivanpal** [comment]
@sgan81 It really does just seem like the README needs updating; the binaries aren't even stored in `bin`, contrary to what the README says. Rather, they just end up directly under `build`, e.g. `build/apfs-fuse`.

**Comment by LithiumH** [comment]
For me installing `libfuse3-dev` installs the FUSE3 headers and I was able to successfully compile the project

**Comment by changeling** [comment]
Just hit this problem. It's a bit of a Catch 22, as the README states Fuse 3 is required for 32-bit, but, at least with Ubuntu, Bionic is the last release for 32-bit _and_ Fuse 3 doesn't seem to be available for bionic. Has anyone found a way through this? (Going to look into building fuse 3, but hoping someone's already packaged it.)

**Comment by meiying-ghost** [comment]
apt-get install libfuse3-dev

**Comment by changeling** [comment]
@meiying-ghost "...Fuse 3 doesn't seem to be available for bionic..."

Or rather, `libfuse3-dev` is not an available package for bionic. i did get fuse 3 to build from source, but it's not simply an 'apt install' thing on bionic or previous 32-bit Ubuntu. Beyond 18.n, 32-bit isn't supported at all. It might be orth updating the README to state that on Ubuntu 32-bit, fuse3 must be built from source.

**Comment by jivanpal** [comment]
@changeling The `libfuse-dev` package is [present in Ubuntu's `main` repo](https://packages.ubuntu.com/bionic/libfuse-dev). A simple `sudo apt install libfuse-dev` should work. If it doesn't, you can download a `.deb` package for i386 [here](https://packages.ubuntu.com/bionic/i386/libfuse-dev/download).

**Comment by changeling** [comment]
@jivanpal  This is a libfuse3 problem. That package doesn’t provide fuse3/fuse.h:

https://packages.ubuntu.com/bionic/i386/libfuse-dev/filelist

**Comment by tuklusan** [comment]
> You couldn't configure it for fuse 2? In the build directory, do the following:
> 
> 1. Execute `cmake ..`
> 2. Execute `ccmake .`
> 3. Navigate to `USE_FUSE3`, and if it is not set to `OFF`, hit enter
> 4. Press `c` to configure
> 5. Press `g` to generate Makefile and exit ccmake
> 6. Execute `make`

Bingo! Thanks Simon.

**Comment by Xara** [comment]
Libfuse3 is in Ubuntu disco and Debian buster

**Comment by americo677** [comment]
> Either use FUSE 2 (configure with ccmake .) or install FUSE 3, probably something like libfuse3-dev

Thank you sgan81!, Installing FUSE 3 with sudo apt install libfuse3-dev, worked out for me!.  Great contribution this project!.  Congratulations and thank you!.

**Comment by gregbenner** [comment]
Ubuntu 20.04
needed `sudo apt install libfuse3-dev` and `sudo apt-get install libbz2-dev`

**Comment by B1QUAD** [comment]
This worked for me (Ubuntu/Pop! OS 20.10):
```bash
sudo apt install fuse3 && sudo apt install libfuse3-dev
```
